### PR TITLE
Assignment4

### PR DESCRIPTION
**Pull request by rutaihwa**
Basic authentication #118 , assignment 4

### CODE REVIEW

File: lib/private.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        return callback(null, false);
+    }
+
+    return callback(null, true, user);
+};
+
+internals.response = function (username) {
+
+    return '<h2>Welcome ' + username + '</h2>';
+
```
Comment (AdriVanHoudt) [inline lib/private.js]: no need for newline here

**Comment by AdriVanHoudt** [comment]
```js
Linting results:
	lib/private.js:
		Line 42: space-before-function-paren - Missing space before function parentheses.
	test/private.js:
		Line 72: space-before-function-paren - Missing space before function parentheses.
```

File: package.json [hunk 2]
```diff
@@ -19,6 +19,7 @@
   "homepage": "https://github.com/hueniverse/hueniversity",
   "dependencies": {
     "hapi": "8.x.x",
+    "hapi-auth-basic": "^2.0.0",
```
Comment (AdriVanHoudt) [inline package.json]: try using the 2.x.x notation for module versions

File: test/private.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
+// Load modules
+
+var Basic = require('hapi-auth-basic');
+var Code = require('code');
+var Lab = require('lab');
+var Lib = require('../lib');
```
Comment (AdriVanHoudt) [inline test/private.js]: call it Hueniversity not Lib, makes it way clearer, see https://github.com/hueniverse/hueniversity/commit/13b9f07922ab36fe4dde0479638d79baf6f099e5#commitcomment-10797003

Comment (rutaihwa) [inline test/private.js]: @AdriVanHoudt , thanks :smile: How come i don't these linting results on my code when i ran the tests?

Comment (AdriVanHoudt) [inline test/private.js]: Do you have a eslintrc file somewhere in the tree? Lab will use that one first and probably you don't have the same rules enabled in that one

Comment (rutaihwa) [inline test/private.js]: No, i dont have .eslintrc file..

Comment (AdriVanHoudt) [inline test/private.js]: hmm latest version of Lab?

Comment (rutaihwa) [inline test/private.js]: I have Lab, not globally installed

**Pull request closed without merging**